### DISCUSSION TITLE
Null value warning for default value of an enum through a generic

### DISCUSSION THREAD

**Discussion by ebfortin** [dotnet/csharplang#6004] [Q&A] [answered]
I have this code:

`public class GammaDistributionModel<K>
	where K : Enum
{
	K _shapeKind = default;`

I get a warning for null value on default. However if I replace K with a valid Enum, I don't get that warning. I was expecting the compiler to infer that since K must be an enum then its default value is 0, it's a value type.

What am I missing?

**Comment by HaloFour** [comment, marked as answer]
A weird quirk of the runtime is that `System.Enum` itself is not a value type and it is also a legal generic type argument to that class.  To fix that I suggest also adding `struct` as a generic constraint.

**Reply by FaustVX** [reply]
Yes to prove that:
```csharp
Enum e = null;
e = E.A;

enum E
{
    A
}
```
[sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCeAfAAQCYBGAWACgBRGAVzAAJUGBeBugGw4G5Lm2qAOgCCvCnzqMqlAN6UGChsMoBfIA==)
